src/EtagGeneratorInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace ZF\HttpCache;
+
+use Zend\Http\Response as HttpResponse;
+
+interface EtagGeneratorInterface
+{
+    /**
+     * Returns an ETag for the given response.
+     *
+     * @param HttpResponse $routeMatch
+     * @return string Etag
+     */
+    public function generateEtag(HttpResponse $response);
```

Comment:
One could want to create the Etag hash from the HTTP request (often simplier than generating the response to create the tag), thus, I would add the HTTP request too.